ExampleApp/ExampleApp/Common/Services/RxBluetoothKitService.swift

```diff
@@ -0,0 +1,137 @@
+        return disconnectionSubject.asObservable()
+    }
+
+    var errorOutput: Observable<Error> {
+        return errorSubject.asObservable()
+    }
+
+    // MARK: - Private subjects
+
+    private let scanningSubject: PublishSubject<ScannedPeripheral> = PublishSubject()
+
+    private let servicesSubject: PublishSubject<[Service]> = PublishSubject()
+
+    private let disconnectionSubject: PublishSubject<Disconnection> = PublishSubject()
+
+    private let errorSubject: PublishSubject<Error> = PublishSubject()
+
+    // MARK: - Private fields
+    private let centralManager: CentralManager = CentralManager(queue: .main)
+
+    private let scheduler: ConcurrentDispatchQueueScheduler
+
+    private let disposeBag = DisposeBag()
+
+    private var connectedPeripherals: [Peripheral] = []
+
+    private var scanningDisposable: Disposable!
+
+    // MARK: - Initialization
+    init() {
+        let timerQueue = DispatchQueue(label: Constant.Strings.defaultDispatchQueueLabel)
+        scheduler = ConcurrentDispatchQueueScheduler(queue: timerQueue)
+    }
+
+
+    // MARK: - Scanning for peripherals
+
+    // You start from observing state of your CentralManager object. Within RxBluetoothKit v.5.0, it is crucial
+    // that you use .startWith(:_) operator, and pass the initial state of your CentralManager with
+    // centralManager.state.
+    func startScanning() {
+        scanningDisposable = centralManager.observeState()
+                .startWith(centralManager.state)
+                .filter {
+                    $0 == .poweredOn
+                }
+                .subscribeOn(MainScheduler.instance)
+                .timeout(4.0, scheduler: scheduler)
+                .flatMap { [unowned self] _ -> Observable<ScannedPeripheral> in
+                    return self.centralManager.scanForPeripherals(withServices: nil)
+                }.bind(to: scanningSubject)
+    }
+    // If you wish to stop scanning for peripherals, you need to dispose the Disposable object, that is created when
+    // you either subscribe for events from an observable returned by centralManager.scanForPeripherals(:_), or you bind
+    // it to an observer. Check starScanning() above for details.
+    func stopScanning() {
+        scanningDisposable.dispose()
+    }
+
+
+    // MARK: - Discovering Services
+
+    // When you discover a service, first you need to establish a connection with a peripheral. Then you call
+    // discoverServices(:_) that peripheral object.
+    func discoverServices(for peripheral: Peripheral) {
+        centralManager.establishConnection(peripheral)
+                .do(onNext: { [unowned self] _ in
+                    self.addConnected(peripheral)
+                    self.observeDisconnect(for: peripheral)
+                })
+                .flatMap {
+                    $0.discoverServices(nil)
+                }.bind(to: servicesSubject)
+                .disposed(by: disposeBag)
```

Comment:
This causes that there is no way to disconnect from the Peripheral in the app. In version 5.0 to disconnect from the Peripheral you need to dispose a Disposable that you get from `establishConnection` method. If you add it to the DisposeBag, that is never disposed, connection will not cancel. I think we want to be possible to disconnect from the Peripheral in the app.

Reply:
Done